custom_logging.py

```diff
@@ -161,15 +161,31 @@ def setup_logger(sentry_url, log_file=None):
     # before this will be enabled.
     settings = QtCore.QSettings()
     flag = settings.value('stream-feature-extractor/sentry-logging', False)
-    if 'SENTRY' in os.environ or flag:
-        client = Client(sentry_url)
-        sentry_handler = SentryHandler(client)
-        sentry_handler.setFormatter(formatter)
-        sentry_handler.setLevel(logging.ERROR)
-        if add_logging_handler_once(logger, sentry_handler):
-            logger.debug('Sentry logging enabled')
+
+    # This test is done because the setting for the value is string sometimes
```

Comment:
the below logic can be simplified by using `type` parameter on the QgsSettings value function 
sample code
```suggestion
   settings = qgis.core.QgsSettings()
   flag = settings.value("stream-feature-extractor/sentry-logging", False, type=bool):
    if flag:
          # configure sentry
    else:
         logger.debug('Sentry logging disabled')
    
```